### PR TITLE
feat(build): add mojo copy-provided-pages

### PR DESCRIPTION
**Pull request by akantcheff**
Add Maven Mojo that:
 * detects if Bonita provided pages (i.e. pages from User Application) are used in project applications
 * resolves used pages (i.e. downloads artifacts from a repository)
 * copies pages artifacts to an output folder (to be packaged afterward)

**Prerequisites: user pages artifacts must be published on Maven Central**

### CODE REVIEW

File: plugin/src/it/copy-provided-pages/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  <packaging>pom</packaging>
+  <modules>
+    <module>app</module>
+  </modules>
+  <properties>
+    <maven.compiler.release>11</maven.compiler.release>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <bonita-project-maven-plugin.version>@project.version@</bonita-project-maven-plugin.version>
+    <bonita.runtime.version>8.0.0</bonita.runtime.version>
```
Comment (akantcheff) [inline plugin/src/it/copy-provided-pages/pom.xml:18]: As we will publish user pages artifacts from 9.0.0, what should we do with this test? Using 9.0.0-SNAPSHOT ?

File: plugin/src/main/java/org/bonitasoft/plugin/build/CopyProvidedPagesMojo.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    protected List<DefaultArtifactCoordinate> detectProvidedPages() throws MojoFailureException {
+        var appSourceDir = project.getBasedir().toPath().resolve(APPLICATION_SOURCE_DIR);
+        if (!Files.exists(appSourceDir) || !Files.isDirectory(appSourceDir)) {
+            log.debug("Applications source directory [{}] does not exist or is not a directory", appSourceDir);
+            return Collections.emptyList();
+        }
+
+        log.debug("Detecting provided pages usage in applications...");
```
Comment (rbioteau) [inline plugin/src/main/java/org/bonitasoft/plugin/build/CopyProvidedPagesMojo.java:127]: Use `info` ?

File: plugin/src/main/java/org/bonitasoft/plugin/build/CopyProvidedPagesMojo.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                .orElseThrow(() -> new NoSuchElementException(
+                        String.format("Missing declaration of dependency: %s", artifactCoordinate)));
+        artifactCoordinate.setVersion(version);
+        return artifactCoordinate;
+    }
+
+    private void copyArtifact(ArtifactCoordinate artifactCoordinate) throws MojoFailureException {
+        var artifact = resolveArtifact(artifactCoordinate);
+        try {
+            Files.copy(artifact.toPath(), outputFolder.toPath().resolve(artifact.getName()));
```
Comment (rbioteau) [inline plugin/src/main/java/org/bonitasoft/plugin/build/CopyProvidedPagesMojo.java:176]: Add `info` log with target destination ?

File: plugin/src/it/copy-provided-pages/pom.xml (hunk 1 above)
Comment (rbioteau) [reply plugin/src/it/copy-provided-pages/pom.xml:18]: I recommend testing it in bonita-project in the meanwhile.

Comment (akantcheff) [reply plugin/src/it/copy-provided-pages/pom.xml:18]: I've done it partially.
But it is strange to have a test on code that is not in the same repository.

Comment (rbioteau) [reply plugin/src/it/copy-provided-pages/pom.xml:18]: I suggest waiting for 9.0.0 release and adding the test at this level later on.
The other solution is to artificially install fake zip archives with the expected coordinate in a first invocation in the `invoker.properties` file, like this:

```properties
invoker.name.1 = Install fake page
# page.zip is an archive file at the root of the test project
invoker.goals.1 = install:install-file -Dfile=page.zip -Dartifact=org.bonitasoft.web.page:page-user-task-list:8.0.0:zip
invoker.buildResult.1 = success
invoker.name.2 = Test copying provided pages
invoker.goals.2 = ${project.groupId}:${project.artifactId}:${project.version}:copy-provided-pages
invoker.buildResult.2 = success

```

Comment (akantcheff) [reply plugin/src/it/copy-provided-pages/pom.xml:18]: Yes it can be a temporary solution waiting for the 9.0.0 release

**Pull request merged**